pkg/build/strategies/sti/sti.go

```diff
@@ -304,21 +338,25 @@ func (builder *STI) Prepare(config *api.Config) error {
 			}
 		}
 		if failedCount == len(requiredAndOptional) {
+			builder.result.BuildInfo = updater.UpdateFailureReason(builder.result.BuildInfo, api.ReasonArtifactsFetchFailed)
 			return fmt.Errorf("Could not download any scripts from URL %v", config.ScriptsURL)
 		}
 	}
 
 	for _, r := range requiredAndOptional {
-		if r.Error == nil {
-			builder.externalScripts[r.Script] = r.Downloaded
-			builder.installedScripts[r.Script] = r.Installed
-			builder.scriptsURL[r.Script] = r.URL
-		} else {
+		if r.Error != nil {
 			glog.Warningf("Error getting %v from %s: %v", r.Script, r.URL, r.Error)
+			continue
 		}
+
+		builder.externalScripts[r.Script] = r.Downloaded
+		builder.installedScripts[r.Script] = r.Installed
+		builder.scriptsURL[r.Script] = r.URL
 	}
 
-	// see if there is a .s2iignore file, and if so, read in the patterns an then search and delete on
+	// see if there is a .s2iignore file, and if so, read in the patterns an then
+	// search and delete on
+	// NOTE: antipattern?
```

Comment:
1) if you want to raise questions about the code unrelated to the changes you're making, this probably isn't the best way to do it since it's creating a distraction from the primary focus of your PR
2) can you be more specific about your concern?